ships/0025-strategy-parameters-enhancements.md

```diff
@@ -0,0 +1,282 @@
+- Values of a build-arg are sometimes common across builds, for example the version of a tool that is used across builds, for example the version of a base image. Users do not want to update several builds individually, potentially forgetting one and continuing to build with an outdated version. Instead they want re-usable values stored in ConfigMaps and reference its keys in multiple Builds or BuildRuns.
+
+That's where the need for the additional features comes from.
+
+### Goals
+
+- Allow the build strategy author to define if a parameter is a single value or an array.
+- Allow the build strategy author to define a default value also for an array parameter.
+- Provide updated build strategy samples that use arrays so that the build strategy author can understand how to do this.
+- Allow build users to define values for an array parameter using simple strings.
+- Allow build users to define values for a string parameter, or an element of an array parameter, using a Secret key reference.
+- Allow build users to define values for a string parameter, or an element of an array parameter, using a ConfigMap key reference.
+- Allow build users to define a prefix for the Secret value.
+- Allow build users to define a prefix for a ConfigMap value.
+
+### Non-Goals
+
+## Proposal
+
+### API Changes
+
+We will extend the Build Strategy Parameter type to support the following:
+
+```yaml
+spec:
+  parameters:
+    - name: build-args
+      description: The values for the ARGs in the Dockerfile.
+      type: array
+      defaults: []
+```
+
+The `type` property is new and allows either `string` or `array` with `string` being the default.
+
+The `defaults` parameter is new and is a string array. No possibility here to reference Secret or ConfigMap values.
+
+The Build and BuildRun parameterValue will be extended to support the following:
+
+```yaml
+spec:
+  paramValues:
+    - name: single-secret-value
+      secretValue:
+        name: secret-name
+        key: secret-key
+    - name: single-configmap-value
+      configMapValue:
+        name: configmap-name
+        key: configmap-key
+    - name: simple-list
+      values:
+        - value: first-item
+        - value: second-item
+    - name: build-args
+      values:
+        - value: KEY1=myarg1
+        - secretValue:
+            name: build-arg-secret
+            key: value
+            format: "KEY2=${SECRET_VALUE}" 
+        - configMapValue:
+            name: build-configuration
+            key: base-image-digest
+            format: "BASE_IMAGE_DIGEST=${CONFIGMAP_VALUE}"
+```
+
+The `secretValue` property is new and can be used instead of value to reference a key in a Secret. Optionally, the `format` property (shown above at the end inside an array) can be used to add text around the Secret value. The scenario is clear there: the value for the build arg comes from a Secret, but the key of the build arg also needs to go there.
```

Comment:
Would it make sense to allow accessing a secret in another namespace, given that the respective current context has access to the other namespace.

Reply:
No, that would be a security problem imo. And implementation-wise it would require a different solution because the `envFrom` pod feature only supports local (= same namespace) references.